pkg/model/awsmodel/autoscalinggroup.go

```diff
@@ -450,17 +450,13 @@ func (b *AutoscalingGroupModelBuilder) buildAutoScalingGroupTask(c *fi.ModelBuil
 
 	for _, extLB := range ig.Spec.ExternalLoadBalancers {
 		if extLB.LoadBalancerName != nil {
-			loadBalancerName := fi.StringValue(extLB.LoadBalancerName)
-			if loadBalancerName != awsup.GetResourceName32(b.Cluster.Name, "api") && loadBalancerName != awsup.GetResourceName32(b.Cluster.Name, "bastion") {
-				loadBalancerName = name + "-" + loadBalancerName
-			}
 			lb := &awstasks.ClassicLoadBalancer{
-				Name:             fi.String(loadBalancerName),
+				Name:             extLB.LoadBalancerName,
 				LoadBalancerName: extLB.LoadBalancerName,
 				Shared:           fi.Bool(true),
 			}
 			t.LoadBalancers = append(t.LoadBalancers, lb)
-			c.AddTask(lb)
+			c.EnsureTask(lb)
```

Comment:
note for reviewers: EnsureTask returns an error if the second invocation provides a task that isn't DeepEqual:

https://github.com/kubernetes/kops/blob/0aa9cf19219b7cb53a1bb96ca7def6b9278188be/upup/pkg/fi/task.go#L70-L74

here, the CLB tasks should always be equal if a CLB is in multiple IGs' externaLoadBalancers because the only task fields that are set use the `LoadBalancerName` and `shared=true`, so this should never return an error.

Reply:
Didn't know about this. May be a good idea to simplify the NLB side as well someday.